examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,294 @@
+    entries.push({
+      fullUrl: generateId(),
+      request,
+      resource: {
+        resourceType: 'Condition',
+        subject: encounter.subject as Reference<Patient>,
+        encounter: { reference: getReferenceString(encounter) },
+        recorder: { reference: getReferenceString(user) },
+        asserter: { reference: getReferenceString(user) },
+        code: conditionData.reasonForVisit,
+        category: [
+          {
+            coding: [
+              {
+                system: 'http://hl7.org/fhir/ValueSet/condition-category',
+                code: 'problem-list-item',
+                display: 'Problem List Item',
+              },
+            ],
+          },
+        ],
+      },
+    });
+  }
+
+  return entries;
+}
+
+/**
+ * This function takes ClinicalImpression data and creates a bundle entry so that it can be added to a batch transaction. The
+ * ClinicalImpression resource represents any notes on an encounter in this context.
+ *
+ * @param clinicalImpressionData - Data object containing codes and values for the ClinicalImpression resources
+ * @param encounter - The encounter that the data is derived from
+ * @param user - The user creating the ClinicalImpressions
+ * @returns A bundle entry with the ClinicalImpression resource that can be used in a batch transaction
+ */
+export function createClinicalImpressionEntry(
+  clinicalImpressionData: ClinicalImpressionData,
+  encounter: Encounter,
+  user: Practitioner
+): BundleEntry | undefined {
+  if (!clinicalImpressionData.assessment) {
+    return undefined;
+  }
+
+  // Create the clinical impression
+  const clinicalImpression: ClinicalImpression = {
+    resourceType: 'ClinicalImpression',
+    status: 'in-progress',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    assessor: { reference: getReferenceString(user) },
+    note: [{ text: clinicalImpressionData.assessment }],
+  };
+
+  // Return the clinical impression in a bundle entry
+  return {
+    fullUrl: generateId(),
+    request: { method: 'PUT', url: `ClinicalImpression?encounter=${getReferenceString(encounter)}` },
+    resource: clinicalImpression,
+  };
+}
+
+/**
+ * This function calculates the BMI of a patient based on their height and weight. Reference: https://my.clevelandclinic.org/health/articles/9464-body-mass-index-bmi
+ *
+ * @param height - The height of the patient
+ * @param weight - The weight of the patient
+ * @returns The BMI of the patient
+ */
+export function calculateBMI(height: Quantity, weight: Quantity): Quantity {
+  if (!height?.value || !weight?.value) {
+    throw new Error('All values must be provided');
+  }
+  const heightM = getHeightInMeters(height);
+  const weightKg = getWeightInKilograms(weight);
+
+  const bmi = Math.round((weightKg / heightM ** 2) * 10) / 10;
+  return {
+    value: bmi,
+    unit: 'kg/m^2',
+  };
+}
+
+function getWeightInKilograms(weight: Quantity): number {
```

Comment:
can you break out all the BMI related functions into a new file called `observation-utils`

Conceptually, these are different from the functions above because they deal with computing single Observation values, rather than full bundles